storm-core/src/jvm/org/apache/storm/daemon/supervisor/Supervisor.java

```diff
@@ -0,0 +1,196 @@
+            localizer.startCleaner();
+
+            EventManagerImp syncSupEventManager = new EventManagerImp(false);
+            EventManagerImp syncProcessManager = new EventManagerImp(false);
+
+            SyncProcessEvent syncProcessEvent = null;
+            if (ConfigUtils.isLocalMode(conf)){
+                localSyncProcess.init(supervisorData);
+                syncProcessEvent = localSyncProcess;
+            }else{
+                syncProcessEvent = new SyncProcessEvent(supervisorData);
+            }
+
+            SyncSupervisorEvent syncSupervisorEvent = new SyncSupervisorEvent(supervisorData, syncProcessEvent, syncSupEventManager, syncProcessManager);
+            UpdateBlobs updateBlobsThread = new UpdateBlobs(supervisorData);
+            RunProfilerActions runProfilerActionThread = new RunProfilerActions(supervisorData);
+
+            if ((Boolean) conf.get(Config.SUPERVISOR_ENABLE)) {
+                StormTimer eventTimer = supervisorData.getEventTimer();
+                // This isn't strictly necessary, but it doesn't hurt and ensures that the machine stays up
+                // to date even if callbacks don't all work exactly right
+                eventTimer.scheduleRecurring(0, 10, new EventManagerPushCallback(syncSupervisorEvent, syncSupEventManager));
+
+                eventTimer.scheduleRecurring(0, Utils.getInt(conf.get(Config.SUPERVISOR_MONITOR_FREQUENCY_SECS)),
+                        new EventManagerPushCallback(syncProcessEvent, syncProcessManager));
+
+                // Blob update thread. Starts with 30 seconds delay, every 30 seconds
+                supervisorData.getBlobUpdateTimer().scheduleRecurring(30, 30, new EventManagerPushCallback(updateBlobsThread, syncSupEventManager));
+
+                // supervisor health check
+                eventTimer.scheduleRecurring(300, 300, new SupervisorHealthCheck(supervisorData));
+
+                // Launch a thread that Runs profiler commands . Starts with 30 seconds delay, every 30 seconds
+                eventTimer.scheduleRecurring(30, 30, new EventManagerPushCallback(runProfilerActionThread, syncSupEventManager));
+            }
+            LOG.info("Starting supervisor with id {} at host {}.", supervisorData.getSupervisorId(), supervisorData.getHostName() );
+            supervisorManger = new SupervisorManger(supervisorData, syncSupEventManager, syncProcessManager);
+        } catch (Throwable t) {
+            if (Utils.exceptionCauseIsInstanceOf(InterruptedIOException.class, t)) {
+                throw t;
+            } else if (Utils.exceptionCauseIsInstanceOf(InterruptedException.class, t)) {
+                throw t;
+            } else {
+                LOG.error("Error on initialization of server supervisor: {}", t);
+                Utils.exitProcess(13, "Error on initialization");
+            }
+        }
+        return supervisorManger;
+    }
+
+    /**
+     * start distribute supervisor
+     */
+    private void launch() {
```

Comment:
If we cannot make it static, that is fine, but it still needs to take an ISupervisor instance as a parameter somewhere.